dell/src/main/java/org/apache/iceberg/dell/ecs/EcsCatalog.java

```diff
@@ -0,0 +1,511 @@
+    }
+  }
+
+  @Override
+  protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
+    return new EcsTableOperations(String.format("%s.%s", catalogName, tableIdentifier),
+        tableURI(tableIdentifier), fileIO, this);
+  }
+
+  @Override
+  protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
+    String tableName = tableIdentifier.name();
+    StringBuilder sb = new StringBuilder();
+
+    sb.append(warehouseLocation).append('/');
+    for (String level : tableIdentifier.namespace().levels()) {
+      sb.append(level).append('/');
+    }
+    sb.append(tableName);
+
+    return sb.toString();
+  }
+
+  /**
+   * Iterate all table objects with the namespace prefix.
+   */
+  @Override
+  public List<TableIdentifier> listTables(Namespace namespace) {
+    if (!namespace.isEmpty() && !namespaceExists(namespace)) {
+      throw new NoSuchNamespaceException("Namespace %s does not exist", namespace);
+    }
+
+    String marker = null;
+    List<TableIdentifier> results = Lists.newArrayList();
+    EcsURI prefix = namespacePrefix(namespace);
+    do {
+      ListObjectsResult listObjectsResult = client.listObjects(
+          new ListObjectsRequest(prefix.bucket())
+              .withDelimiter("/")
+              .withPrefix(prefix.name())
+              .withMarker(marker));
+      marker = listObjectsResult.getNextMarker();
+      results.addAll(listObjectsResult.getObjects().stream()
+          .filter(s3Object -> s3Object.getKey().endsWith(TABLE_OBJECT_SUFFIX))
+          .map(object -> parseTableId(namespace, prefix, object))
+          .collect(Collectors.toList()));
+    } while (marker != null);
+
+    LOG.debug("Listing of namespace: {} resulted in the following tables: {}", namespace, results);
+    return results;
+  }
+
+  /**
+   * Get object prefix of namespace.
+   */
+  private EcsURI namespacePrefix(Namespace namespace) {
+    String prefix;
+    if (namespace.isEmpty()) {
+      prefix = warehouseLocation.name();
+    } else {
+      prefix = String.format("%s%s/", warehouseLocation.name(),
+          String.join("/", namespace.levels()));
+    }
+
+    return new EcsURI(warehouseLocation.bucket(), prefix);
+  }
+
+  private TableIdentifier parseTableId(Namespace namespace, EcsURI prefix, S3Object s3Object) {
+    String key = s3Object.getKey();
+    Preconditions.checkArgument(key.startsWith(prefix.name()),
+        "List result should have same prefix", key, prefix);
+
+    String tableName = key.substring(
+        prefix.name().length(),
+        key.length() - TABLE_OBJECT_SUFFIX.length());
+    return TableIdentifier.of(namespace, tableName);
+  }
+
+  /**
+   * Remove table object. If the purge flag is set, remove all data objects.
+   */
+  @Override
+  public boolean dropTable(TableIdentifier identifier, boolean purge) {
+    if (!tableExists(identifier)) {
+      throw new NoSuchTableException("Table %s does not exist", identifier);
+    }
```

Comment:
As the `dropTable` 's interface definition,  we should return `false` rather than throwing a NoSuchTableException.

```java
/**
   * Drop a table and delete all data and metadata files.
   *
   * @param identifier a table identifier
   * @return true if the table was dropped, false if the table did not exist
   */
  default boolean dropTable(TableIdentifier identifier) {
    return dropTable(identifier, true /* drop data and metadata files */);
  }
```

Reply:
Fixed.